src/XrdCeph/XrdCephPosix.cc

```diff
@@ -706,8 +707,23 @@ int ceph_posix_open(XrdOucEnv* env, const char *pathname, int flags, mode_t mode
        logwrapper((char*)"Could not find size or stripe_unit xattr for %s", fr.name.c_str());
       }
      else{
-       fr.stripeUnit = std::stoull(d_stripeUnit.c_str());
-       fr.objectSize = std::stoull(d_objectSize.c_str());
+       //librados's c_str() method does not return a NULL-terminated string, hence why we need to cleanup here
+       char cleanStripeUnit[1024];
+       char cleanObjectSize[1024];
+       unsigned int stripeUnitLength = std::min((unsigned int)1023, d_stripeUnit.length());
+       unsigned int objectSizeLength = std::min((unsigned int)1023, d_objectSize.length());
+       strncpy( cleanStripeUnit, d_stripeUnit.c_str(), stripeUnitLength );
+       strncpy( cleanObjectSize, d_objectSize.c_str(), objectSizeLength );
+       cleanStripeUnit[stripeUnitLength] = '\0';
+       cleanObjectSize[objectSizeLength] = '\0';
+       if(fr.stripeUnit != std::stoull(cleanStripeUnit)){
+         logwrapper((char*)"WARNING: stripe unit does not match defaults. object size is %s",cleanStripeUnit);
+       }
+       if(fr.objectSize != std::stoull(cleanObjectSize)){
+         logwrapper((char*)"WARNING: object size does not match defaults. object size is %s",cleanObjectSize);
```

Comment:
I'd suggest to also add file name here, e.g. something like "..object size is %s, file name: %s"... .